pkg/vulnsrc/oracle-oval/oracle-oval.go

```diff
@@ -216,32 +217,40 @@ type archFlavor struct {
 	Flavor PkgFlavor
 }
 
-// mergeAdvisoriesEntries merges advisories by picking the latest version for each arch+flavor.
-// There may be multiple advisories that fix the same vulnerability, possibly providing multiple fixed versions.
-// In this case, we need to determine the "latest" version, which is now defined as the highest (greatest) version number.
+// mergeAdvisoriesEntries merges advisories by selecting the latest version for each combination of architecture and flavor.
+// The "latest" version is determined as the highest (greatest) version number for each (arch, flavor) pair.
 //
-// Additionally, we choose a single fixed version for backward compatibility, which is derived from
-// the highest normal flavor version on the x86_64 architecture if available.
+// The process is as follows:
+// 1. Select the latest version for each (arch, flavor) combination.
+// 2. Aggregate architectures for each selected version into a single entry, grouping them by their version.
+// 3. Build the final list of advisories, sorted by version in ascending order.
+//
+// Additionally, for backward compatibility, we determine a single primary fixed version.
+// This is derived from the highest normal flavor version on the x86_64 architecture if available.
 func mergeAdvisoriesEntries(advisories types.Advisories) types.Advisories {
 	// Step 1: Select the latest version per (arch, flavor)
 	latestVersions := selectLatestVersions(advisories)
 
-	// Step 2: Aggregate arches by their chosen version
-	versionToArches := make(map[string][]string)
+	// Step 2: Aggregate architectures by their chosen version
+	versionToArches := make(map[string]types.Advisory)
 	for k, v := range latestVersions {
-		versionToArches[v] = append(versionToArches[v], k.Arch)
+		adv := versionToArches[v.FixedVersion]
+		adv.VendorIDs = v.VendorIDs
```

Comment:
It can be wrong for case when 2 advisories with same fixedVersion got from 2 different ELSA-IDs
e.g.
- ELSA-xxxx-0001 - fixedVersion == 0.0.1, arch == amd64
- ELSA-xxxx-0002 - fixedVersion == 0.0.1, arch == aarch64
for this case entry will include only one ELSA-xxxx-0002 (or 0001)

Reply:
perhaps we need to create 2 different entries for this case

Reply:
@DmitriyLewen  Thanks for the review.

As trivy doesn't need this vendor ids data in db, removed them before storing in db so there is no impact on db size.

Addressed same fix version with different elsa-ids and architectures case.

Can you please review this? Thank you!!

Reply:
Hello @santhosh1729 
I am currently bezy on new Trivy release.

I will check your PR after release.